harness/determined/pytorch/_pytorch_context.py

```diff
@@ -412,26 +430,46 @@ def _average_gradients(self, parameters: Any, divisor: int) -> None:
         for p in filter(lambda param: param.grad is not None, parameters):
             p.grad.data.div_(divisor_value)
 
-    def _step_optimizer(
+    def step_optimizer(
         self,
         optimizer: torch.optim.Optimizer,  # type: ignore
         clip_grads: Optional[Callable[[Iterator], None]] = None,
+        auto_zero_grads: bool = True,
     ) -> None:
         """
         Perform a single optimization step.
 
         This function must be called once for each optimizer. However, the order of
-        different optimizers' optimization steps can be specified by calling
-        this function in different orders. Also, gradient accumulation across iterations
-        is performed by the Determined training loop by setting the experiment configuration
-        optimization.aggregation_frequency.
+        different optimizers' steps can be specified by calling this function in different
+        orders. Also, gradient accumulation across iterations is performed by the Determined
+        training loop by setting the experiment configuration field
+        ``optimization.aggregation_frequency``.
+
+        Here is a code example:
+
+        .. code-block:: python
+
+            def clip_grads(params):
+                torch.nn.utils.clip_grad_norm_(params, 0.0001),
+
+            self.context.step_optimizer(self.opt1, clip_grads)
 
         Arguments:
             optimizer(``torch.optim.Optimizer``): Which optimizer should be stepped.
             clip_grads(a function, optional): This function should have one argument for
                 parameters in order to clip the gradients
+            auto_zero_grads(bool, optional): Automatically zero out gradients automatically
+                after stepping the optimizer. If set to be false, you need to call
+                ``optimizer.zero_grad()`` manually. Note that if
+                ``optimizations.aggregation_frequency`` is larger than 1, you can only
+                set it to be true.
```

Comment:
We need to explicitly state that the default is `True`.  Also, rephrase `If set to be false` -> `If false` and `you can only set it to be true` -> `auto_zero_grads must be true`